reports/sla/entrypoint.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021, CloudBlue
+# All rights reserved.
+#
+import datetime
+
+from reports.utils import convert_to_datetime, get_dict_element
+
+
+def generate(
+    client=None,
+    parameters=None,
+    progress_callback=None,
+    renderer_type=None,
+    extra_context_callback=None,
+):
+    """
+    Extracts data from Connect using the ConnectClient instance
+    and input data provided as arguments, applies
+    required transformations (if any) and returns the data rendered
+    by the given renderer on the arguments list.
+    Some renderers may require extra context data to generate the report
+    output, for example in the case of the Jinja2 renderer...
+
+    :param client: An instance of the CloudBlue Connect
+                    client.
+    :type client: cnct.ConnectClient
+    :param input_data: Input data used to calculate the
+                        resulting dataset.
+    :type input_data: dict
+    :param progress_callback: A function that accepts t
+                                argument of type int that must
+                                be invoked to notify the progress
+                                of the report generation.
+    :type progress_callback: func
+    :param renderer_type: Renderer required for generating report
+                            output.
+    :type renderer_type: string
+    :param extra_context_callback: Extra content required by some
+                            renderers.
+    :type extra_context_callback: func
+    """
+    offset_red = int(parameters['offset_red_days'])
+    offset_yellow = int(parameters['offset_yellow_days'])
+
+    generator = client.requests.all().order_by('created').limit(1000)
+
+    total = client.requests.all().count()
+
+    progress = 0
+
+    levels = {
+        'red': offset_red,
+        'yellow': offset_yellow,
+    }
+
+    for record in generator:
+        yield _process_line(record, levels)
+        progress += 1
+        progress_callback(progress, total)
+
+
+def _get_awaiting_for(data):
+    return (datetime.datetime.now() - convert_to_datetime(data['created'])).days
```

Comment:
Probably we should ensure that all datetime calculations are made in the same timezone. 

datetime.datetime.now() returns the local time but if I'm not wrong data['created'] is UTC.
Could you please check ?